src/native/libs/System.Security.Cryptography.Native/opensslshim.h

```diff
@@ -501,6 +507,9 @@ const EVP_CIPHER* EVP_chacha20_poly1305(void);
     REQUIRED_FUNCTION(SSL_read) \
     REQUIRED_FUNCTION(SSL_renegotiate) \
     REQUIRED_FUNCTION(SSL_renegotiate_pending) \
+    REQUIRED_FUNCTION(SSL_SESSION_free) \
+    REQUIRED_FUNCTION(SSL_SESSION_get0_hostname) \
```

Comment:
SSL_SESSION_get0_hostname was added in OSSL111.  Looks like we're still building somewhere with 102 or 110 headers, which makes this a bit more complicated.

Reply:
ok. I can fix that. I think compat  function would do as we have access to the information.